src/core/thread/mle.cpp

```diff
@@ -408,10 +409,22 @@ ThreadError Mle::SetStateChild(uint16_t aRloc16)
 
     if (mPreviousPanId != Mac::kPanIdBroadcast)
     {
-        mPreviousPanId = Mac::kPanIdBroadcast;
-        mNetif.GetAnnounceBeginServer().SendAnnounce(1 << mPreviousChannel);
+        if (mDeviceMode & ModeTlv::kModeFFD)
+        {
+            const MeshCoP::Timestamp *localTimestamp;
+            localTimestamp = mNetif.GetActiveDataset().GetNetwork().GetTimestamp();
+
+            VerifyOrExit(localTimestamp != NULL && localTimestamp->Compare(mPreviousActiveTimestamp) < 0, ;);
```

Comment:
Do we really need to handle this case?  If `mPreviousPanId` is set and we implement this properly, then we should not be rolling back to the Active Timestamp.